rutil/Errdes.cxx

```diff
@@ -0,0 +1,409 @@
+#include <iostream>
+#include <map>
+#include <string>
+#include <errno.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+#include "rutil/Errdes.hxx"
+
+#ifdef _WIN32
+#include <winsock.h>
+#include <windows.h>
+#endif
+
+using namespace std;
+
+#define _GET_CODE(x) #x
+#define ERR_WITH_CODE(m, s) #m ": " s " (" _GET_CODE(m) ")"
+
+#ifdef _WIN32
+map <int, string> WinErrorMsg;
+#elif __linux__
+map <int, string> ErrornoErrorMsg;
+#endif
+map <int, string> OpenSSLErrorMsg;
+map <int, string> X509ErrorMsg;
+
+
+string ErrnoError::SearchErrorMsg(int Error)
+{
+  string result;
+
+#ifdef _WIN32
+    result = WinErrorMsg[Error];
```

Comment:
check if the map contains the key before trying to retrieve the value

please do the same thing for each of the other lookups too